design/ui/pagination/src/Pagination.js

```diff
@@ -0,0 +1,292 @@
+import React, { Fragment } from 'react';
+import pt from 'prop-types';
+import classNames from 'classnames/bind';
+import { noop } from 'lodash';
+
+import { Button, Select, Header, Popup } from 'semantic-ui-react';
+import 'semantic-ui-css-custom';
+import styles from './Pagination.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /** Can render as a different tag or component */
+  as: pt.oneOfType([pt.string, pt.node]),
+  /** the items collection length */
+  count: pt.number,
+  /** entity name in plural form */
+  entityNamePlural: pt.string,
+  /** entity name in singular form */
+  entityNameSingular: pt.string,
```

Comment:
I wonder if there is a way for us to reuse the plurals logic in our translations and duplicate that here. That way we can just pass a single entity object from our translations and have it work without us having to separate it out. It would also reduce the API surface area.

What are your thoughts on this? If we went that route, maybe the object signature would be 1:1 the same as how we define a plural in POEditor? That would also give better localization support since some languages the plural changes based on the count, like Russian has words where it has singular, plural 2-9, plural 10+

Reply:
I am pretty sure we can replicate the behavior you mention for this. Also, I was thinking on translations but since I don't know if that's already set up I didin't explore that path.

Let me explore it and share my findings.